gui/config.py

```
@@ -45,11 +46,19 @@ def load_config() -> Dict[str, Any]:
+    truncated — which load_config() would silently discard, losing the
+    user's hf_token, custom_models, theme, and auto_save settings.
+    """
     _ensure_dir()
     try:
-        with open(_CONFIG_FILE, "w", encoding="utf-8") as f:
+        tmp = _CONFIG_FILE.with_suffix(".json.tmp")
+        with open(tmp, "w", encoding="utf-8") as f:
             json.dump(cfg, f, indent=2)
+        os.replace(tmp, _CONFIG_FILE)
     except Exception:
         pass  # best-effort
```

Ruff (0.15.18)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)